enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,404 @@
+- Add structure to controller code to be resilient to missing cloud credentials / not crash.
+
+### Workflow Description
+
+This diagram shows the steps a user will need to take in the proposed flow (specific to STS):
+
+![STS Tomorrow](STS_tomorrow.jpg)
+
+Making a layered operator ready to work on an STS enabled cluster will involve the following steps:
+
+Operator installation under the proposed system:
+1. user goes to console and starts operator install
+2. console detects token-based (STS) cluster, and token-based-supporting operator
+3. console prompts user to create roles+supply ARN-type string
+4. console creates subscription with ARN-type string embedded as a spec.config.env
+5. operator deployment is created with ARN-type string in env
+6. operator creates `CredentialsRequest` including the ARN-type string
+7. Cloud Credential Operator populates Secret based on `CredentialsRequest`
+8. operator loads Secret, makes cloud service requests
+
+
+For the operator author team:
+- Add code to create a CredentialRequests (it will not be stored in the bundle 
+as part of this EP)
+- Add "role-arn"-type fields to the CredentialRequests
+- Add eventing to report status on a CR to indicate lacking STS credentials for fully operational deploy or update.
+
+For the Cloud Credential Operator:
+- Allow detection that a cluster is using time-based tokens and create a Secret containing the ARN + token path per 
+ cloud platform.
+
+For Operator Administrator:
+- Supply cloud credentials for STS (ARN ID, etc) in the known location.
+
+For OperatorHub, Console:
+- Prompt for missing credentials that prevent fully operational install on STS cluster. Same for upgrades.
+- Badge indicating "STS enabled" (this is a naming suggestion to avoid implying that non-badged operators not needing
+ cloud provider resource access would not work) operators.
+
+This graph shows the system as it needs to work (AWS only), not the proposed solution workflow:
+```mermaid
+graph  LR
+  sa([ServiceAccount Signing Keys]) -- Public <br> Key -->s3[S3 Bucket w/ OIDC Config];
+  sa([ServiceAccount Signing Keys]) -- Private <br> Key -->token_signing;
+  service<-.->config
+  service<-.->Pod
+  Pod-->role
+  subgraph cluster [AWS]
+  s3;
+  iam_id[IAM Identity Provider] -- Trust Tokens<br>Signed -->s3;
+  role[IAM Role For Pod]<-->service[AWS Services];
+  end
+  subgraph cluster2 [OpenShift]
+  token_signing[Token Signing]-->projected[Projected<br>ServiceAccount<br>Token];
+  subgraph Pod
+  config[AWS Config File]-->projected
+  end
+  end
+  classDef plain fill:#ddd,stroke:#fff,stroke-width:4px,color:#000;
+  classDef k8s fill:#326ce5,stroke:#fff,stroke-width:4px,color:#fff;
+  classDef cluster fill:#fff,stroke:#bbb,stroke-width:2px,color:#326ce5;
+  classDef cluster2 fill:#fff,stroke:#bbb,stroke-width:2px,color:#326ce5;
+  class config,role,service,s3,projected,token_signing,iam_id,pod1,pod2 k8s;
+  class sa plain;
+  class cluster cluster;
+  class cluster2 cluster1;
```

Comment:
This seems like valuable feedback for overall STS implementation but does this directly affect the changes being proposed in this enhancement? If not lets keep the focus of this enhancement on the OLM and Operator interactions here but find somewhere else to follow up on AWS STS improvements.

Reply:
yeah i almost suggested removing this diagram entirely as very little of it is really relevant to what's being proposed here.  I decided not to suggest it since it's "helpful reference information", but this EP shouldn't be the canonical source of how STS/OIDC provider setup works.

Reply:
>  but does this directly affect the changes being proposed in this enhancement? If not

You are right, does not affect this proposal.

Commented as informational for posterity when reading `the system as it needs to work (AWS only)`.